### DISCUSSION TITLE
We want your feedback ~ TraitsDB

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/mdn-community#489] [Content]
We recently merged the TraitsDB pull request opened by @peterbe quite some time ago - https://github.com/mdn/yari/pull/3280

We did this because we believe this tool could be useful but, we want _your_ feedback. Below is a video explaining how to find and use the basics of the tool.

https://user-images.githubusercontent.com/10350960/161932946-53df7836-e589-4867-be98-bb6376349349.mov

## References

- https://www.npmjs.com/package/alasql
- https://github.com/mdn/yari/#quickstart

**Comment by peterbe** [comment]
I would actually recommend that you use [docsQL](https://github.com/peterbe/docsql) instead. 
It was based on this work but much more mature and probably a lot faster too. 
However, I think the Traits DB in Yari itself already has some plugins to deal with KumaScript macros. 

For the [blog post announcement](https://www.peterbe.com/plog/introducing-docsql) I actually set up [an instance based on mdn/content](https://docsql-demo-mdn.peterbe.com/?query=SELECT+title%2C+_url%2C+wordCount+FROM+%3F+WHERE+%27JavaScript%27+IN+tags+ORDER+BY+3+DESC+LIMIT+10) but without any plugins.

**Reply by schalkneethling** [reply]
Thanks, Peter! I assumed as much 😄 but, wanted to get an idea of what people thought about this kind of tool in general.

**Comment by hamishwillee** [comment]
My first thought was that now we (@wbamberg) are starting to look at structure and behaviour associated with page types, this might be a useful part of the story for validating that certain markdown structures exist or do not exist in a page. For example - validating that there is a heading like below in every API function doc.
```
## Examples
```
Ideally also reporting on whether it has content or a placeholder, and catching cases like "Example"

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant.
